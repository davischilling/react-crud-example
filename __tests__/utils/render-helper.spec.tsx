import { BrowserRouter } from 'react-router-dom';
import { renderWithProviders } from './render-helper';

describe('RenderWithProviders', () => {
  test('renders children with providers', () => {
    const { asFragment } = renderWithProviders<{
      mockState: any;
    }>(
      <div>
        <p>Mocked Content</p>
      </div>,
      ['/mocked-path'],
      {
        wrapper: BrowserRouter,
      },
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders children with providers and initial state', () => {
    const { asFragment } = renderWithProviders<{
      mockState: any;
    }>(
      <div>
        <p>Mocked Content</p>
      </div>,
      ['/mocked-path'],
      {
        wrapper: BrowserRouter,
        states: [
          {
            atom: {} as any,
            value: {
              mockState: {
                isLoading: false,
                data: [],
              },
            },
          },
        ],
      },
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('should execute initializeState correctly', () => {
    const set = jest.fn();
    const states = [
      {
        atom: {} as any,
        value: {
          mockState: {
            isLoading: false,
            data: [],
          },
        },
      },
      {
        atom: {} as any,
        value: {
          mockState: {
            isLoading: false,
            data: [],
          },
        },
      },
    ];
    const initializeState = ({ set }: any): void => {
      [...states, { atom: {} as any, value: {} as any }].forEach((state) =>
        set(state.atom, state.value),
      );
    };
    initializeState({ set });
    expect(set).toHaveBeenCalledTimes(3);
    expect(set).toHaveBeenCalledWith(states[0].atom, states[0].value);
    expect(set).toHaveBeenCalledWith(states[1].atom, states[1].value);
  });
});
