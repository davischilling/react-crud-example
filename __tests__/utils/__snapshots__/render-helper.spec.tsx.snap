// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderWithProviders renders children with providers 1`] = `
<DocumentFragment>
  <div>
    <p>
      Mocked Content
    </p>
  </div>
</DocumentFragment>
`;

exports[`RenderWithProviders renders children with providers and initial state 1`] = `
<DocumentFragment>
  <div>
    <p>
      Mocked Content
    </p>
  </div>
</DocumentFragment>
`;
